--- app/code/community/TBT/Testsweet/Model/Observer/Crontest-o.php	2017-10-10 10:36:28.000000000 +0200
+++ app/code/community/TBT/Testsweet/Model/Observer/Crontest.php	2017-10-10 10:53:05.000000000 +0200
@@ -6,11 +6,53 @@
         $timestamp = $this->getCurrentTimestamp();
         Mage::getConfig()
             ->saveConfig('testsweet/crontest/timestamp', $timestamp, 'default', 0)
-            ->reinit();
+            ;
+        // PATCH
+        // DO NOT REINIT FULL CACHE! Instead set flag that config cache needs to be invalidated.
+        //    ->reinit();
+        //~PATCH
 
         return $this;
     }
     
+    /**
+     * @return Varien_Db_Adapter_Pdo_Mysql
+     */
+    protected function _getDatabaseRead()
+    {
+        return Mage::getSingleton('core/resource')->getConnection('core_read');
+    }
+
+    protected function _getDirectStoreConfig($path, $store = null, $backendModel = null)
+    {
+        /** @var Mage_Core_Model_Resource_Config $configResource */
+        $configResource = Mage::getConfig()->getResourceModel();
+        $scopeId = $store === 0 ? $store : (int)Mage::app()->getStore($store)->getId();
+        if ($scopeId) {
+            $connection = $this->_getDatabaseRead();
+            $select = $connection->select();
+            $select->from($configResource->getMainTable(), array('value'));
+            $select->where('scope = :scope AND scope_id = :scope_id AND path = :path');
+
+
+            $value = $connection->fetchOne($select, array(
+                'scope' => $scopeId === 0 ? 'default' : 'stores',
+                'scope_id' => $scopeId,
+                'path' => $path,
+            ));
+
+            if ($backendModel && !empty($value)) {
+                $backend = Mage::getModel($backendModel);
+                $backend->setPath($path)->setValue($value)->afterLoad();
+                $value = $backend->getValue();
+            }
+
+            return $value;
+        }
+
+        return null;
+    }
+
     public function getCurrentTimestamp() {
         $timestamp = (string)time();
         return $timestamp;
@@ -18,7 +60,10 @@
 
     public function getCronTimestamp() {
         //$timestamp = (string)Mage::getConfig()->getNode('testsweet/crontest/timestamp', 'default', 0);
-        $timestamp = Mage::getStoreConfig('testsweet/crontest/timestamp');
+        // PATCH
+        //$timestamp = Mage::getStoreConfig('testsweet/crontest/timestamp');
+        $timestamp = $this->_getDirectStoreConfig('testsweet/crontest/timestamp', 0);
+        //~PATCH
         return $timestamp;
     }
 
--- app/code/community/TBT/Rewards/Helper/Cron.php	Tue Oct 10 12:01:09 2017 +0300
+++ app/code/community/TBT/Rewards/Helper/Cron.php	Tue Oct 10 11:05:50 2017 +0200
@@ -58,7 +58,10 @@
      * @return string timestamp
      */
     public function getCronTimestamp() {
-        $timestamp = Mage::getStoreConfig('testsweet/crontest/timestamp');
+        //PATCH elvin@vaimo take direct value from config table, not cached value
+        //$timestamp = Mage::getStoreConfig('testsweet/crontest/timestamp');
+        $timestamp = Mage::getSingleton('testsweet/observer_crontest')->getCronTimestamp();
+        //~PATCH
         return $timestamp;
     }
 
