--- app/design/frontend/base/default/template/pos/shipping_additional-o.phtml	2017-05-08 08:07:42.000000000 +0000
+++ app/design/frontend/base/default/template/pos/shipping_additional.phtml	2017-05-08 08:21:49.000000000 +0000
@@ -48,13 +48,13 @@
 </div>
 
 <script type="text/javascript">
-
+(function($) {
     var POSStoresData             = <?php echo $this->getStoresJson(); ?>;
-    var POSInStorePickupInput     = $('s_method_bakerloo_store_pickup_bakerloo_store_pickup');
-    var POSInStorePickupContainer = $('pos-store-pickup');
+    var POSInStorePickupInput     = $('#s_method_bakerloo_store_pickup_bakerloo_store_pickup');
+    var POSInStorePickupContainer = $('#pos-store-pickup');
 
     function POSHandleShippingMethodClick(event) {
-        if(this.readAttribute('id') == 's_method_bakerloo_store_pickup_bakerloo_store_pickup') {
+        if($(this).attr('id') == 's_method_bakerloo_store_pickup_bakerloo_store_pickup') {
             POSInStorePickupContainer.show();
         }
         else {
@@ -77,10 +77,10 @@
 
     function POSHandleShippingStorePick(event) {
 
-        $('pos-store-pickup-address').hide();
+        $('#pos-store-pickup-address').hide();
 
-        if(this.tagName.toLowerCase() == 'select') {
             var storeId = '';
+        if(this.tagName.toLowerCase() == 'select') {
             this.descendants().each(function(element, index){
                 if(element.selected) {
                     storeId = element.value.split('_')[1];
@@ -88,36 +88,38 @@
             });
         }
         else {
-            var storeId = this.readAttribute('value').split('_')[1];
+            storeId = this.readAttribute('value').split('_')[1];
         }
 
         var storeAddress = getStoreAddress(storeId);
         if(storeAddress) {
-            $$('div#pos-store-pickup-address p').first().update(storeAddress);
-            $('pos-store-pickup-address').show();
+            $('div#pos-store-pickup-address p:first').html(storeAddress);
+            $('#pos-store-pickup-address').show();
         }
 
     }
 
-    $$('dl.sp-methods input[type=radio]').invoke('observe', 'click', POSHandleShippingMethodClick);
+    //$$('dl.sp-methods input[type=radio]').invoke('observe', 'click', POSHandleShippingMethodClick);
+    $('dl.sp-methods input[type=radio]').click(POSHandleShippingMethodClick);
 
-    if(POSInStorePickupInput.checked) {
+    if(POSInStorePickupInput.is(':checked')) {
         POSInStorePickupContainer.show();
     }
 
-    var POSRadios = $$('input[name=pos_store_pickup_store]');
+    var POSRadios = $('input[name=pos_store_pickup_store]');
 
     if(POSRadios.length) {
         //Radio buttons
-        POSRadios.invoke('observe', 'click', POSHandleShippingStorePick);
+        POSRadios.click(POSHandleShippingStorePick);
     }
     else {
         //Select
-        var mySelect = $('pos-store-pickup-stores');
+        var mySelect = $('#pos-store-pickup-stores');
 
-        if(null != mySelect) {
-            mySelect.observe('change', POSHandleShippingStorePick);
+        if(mySelect.length) {
+            mySelect.change(POSHandleShippingStorePick);
         }
     }
+})(jQuery);
 
 </script>
\ No newline at end of file
