--- app/code/core/Enterprise/PageCache/Model/Container/CatalogProductList-o.php	2013-12-12 15:00:53.000000000 +0200
+++ app/code/core/Enterprise/PageCache/Model/Container/CatalogProductList.php	2013-12-12 14:58:25.000000000 +0200
@@ -55,4 +55,15 @@
 
         return '';
     }
+
+    /**
+     * Retrieve cache id
+     *
+     * @return string
+     */
+    protected function _getCacheId()
+    {
+        return parent::_getCacheId() . $this->_getProductId();
+    }
+
 }
