--- lib/tcpdf/tcpdf-o.php	Thu Apr 28 10:47:24 2016 +0000
+++ lib/tcpdf/tcpdf.php	Fri May 06 19:07:46 2016 +0000
@@ -7022,8 +7022,8 @@
 			if ((method_exists('TCPDF_IMAGES', $mtd)) AND (!($resize AND (function_exists($gdfunction) OR extension_loaded('imagick'))))) {
 				// TCPDF image functions
 				$info = TCPDF_IMAGES::$mtd($file);
-				if (($ismask === false) AND ($imgmask === false) AND (strpos($file, '__tcpdf_'.$this->file_id.'_imgmask_') === FALSE)
-					AND (($info === 'pngalpha') OR (isset($info['trns']) AND !empty($info['trns'])))) {
+				if ((!$ismask) AND (!$imgmask) AND (strpos($file, '__tcpdf_'.$this->file_id.'_imgmask_') === FALSE)
+					AND (($info === 'pngalpha') OR ((isset($info['trns']) AND !empty($info['trns']))))) {
 					return $this->ImagePngAlpha($file, $x, $y, $pixw, $pixh, $w, $h, 'PNG', $link, $align, $resize, $dpi, $palign, $filehash);
 				}
 			}
