--- app/code/core/Mage/Cron/Model/Observer-o.php	2015-06-17 12:19:45.000000000 +0300
+++ app/code/core/Mage/Cron/Model/Observer.php	2015-06-17 12:04:33.000000000 +0300
@@ -330,6 +330,9 @@
                 ->setFinishedAt(strftime('%Y-%m-%d %H:%M:%S', time()));
 
         } catch (Exception $e) {
+            if ($schedule->getJobCode() == 'enterprise_refresh_index') {
+                Mage::logException($e);
+            }
             $schedule->setStatus($errorStatus)
                 ->setMessages($e->__toString());
         }
