--- app/code/core/Mage/Checkout/Model/Type/Onepage.php	(revision )
+++ app/code/core/Mage/Checkout/Model/Type/Onepage.php	(revision )
@@ -498,7 +498,7 @@
                 'lastname'     => 'lastname',
                 'email'        => 'email',
                 'password'     => 'customer_password',
-                'confirmation' => 'confirm_password',
+                'password_confirmation' => 'confirm_password',
                 'taxvat'       => 'taxvat',
                 'gender'       => 'gender',
             ) as $key => $dataKey) {
